components/mongodb/mongodb.app.mjs

```diff
@@ -1,11 +1,234 @@
+      const uri = `mongodb+srv://${username}:${password}@${hostname}/${database}?retryWrites=true&w=majority`;
+      await mongoose.connect(uri);
+    },
+    getCollections() {
+      return new Promise((resolve, reject) => {
+        this.connect()
+          .then(async () => {
+            try {
+              const collections = await mongoose.connection.db.listCollections().toArray();
+              resolve(collections.map((collection) => collection.name).sort());
+            } catch (err) {
+              reject(err);
+            } finally {
+              mongoose.connection.close();
+            }
+          })
+          .catch((err) => reject(err));
+      });
+    },
+    createDocument(data, collection, parseNumbers, parseBooleans, parseDates) {
+      return new Promise((resolve, reject) => {
+        this.connect().then(async () => {
+          try {
+            const Model = mongoose.model(collection, this.getSchemaByData(data));
+            const document = new Model(this.parseStrings(
+              data,
+              parseNumbers,
+              parseBooleans,
+              parseDates,
+            ));
+            await document.save();
+            mongoose.connection.close(() => {
+              setTimeout(() => resolve(document));
+            });
+          } catch (err) {
+            mongoose.connection.close(() => {
+              setTimeout(() => reject(err));
+            });
+          }
+        })
+          .catch((err) => reject(err));
+      });
+    },
+    updateDocument(collection, _id, data, parseNumbers, parseBooleans, parseDates) {
+      return new Promise((resolve, reject) => {
+        this.connect().then(async () => {
+          try {
+            const Model = mongoose.model(collection, this.getSchemaByData(data));
+            await Model.findByIdAndUpdate(_id, this.parseStrings(
+              data,
+              parseNumbers,
+              parseBooleans,
+              parseDates,
+            ));
+            mongoose.connection.close(() => {
+              setTimeout(() => resolve());
+            });
+          } catch (err) {
+            mongoose.connection.close(() => {
+              setTimeout(() => reject(err));
+            });
+          }
+        })
+          .catch((err) => reject(err));
+      });
+    },
+    deleteDocumentById(collection, _id) {
+      return new Promise((resolve, reject) => {
+        this.connect()
+          .then(async () => {
+            try {
+              const schema = new mongoose.Schema(undefined, {
+                strict: false,
+              });
+              const Model = mongoose.model(collection, schema);
+              await Model.deleteOne({
+                _id,
+              });
+              mongoose.connection.close(() => {
+                setTimeout(resolve());
+              });
+            } catch (err) {
+              mongoose.connection.close(() => {
+                setTimeout(reject(err));
+              });
```

Comment:
I get the same error running `delete-a-document`, mentioned in https://github.com/PipedreamHQ/pipedream/pull/2072#issuecomment-1021686503.

<img width="500" alt="delete-a-document" src="https://user-images.githubusercontent.com/19861096/151249287-d301b147-52b6-4b54-ae8b-f4e86024db62.png">


I think the issue is that `setTimeout()`'s first argument must be a function:
```diff
-                setTimeout(resolve());
+                setTimeout(resolve);
```
or
```diff
-                setTimeout(resolve());
+                setTimeout(() => resolve());
```

Reply:
Funny. I did not get these errors, but indeed, this is suppose to be a function and I am passing the function result